siege_utilities/geo/providers/boundary_providers.py

```
@@ -0,0 +1,399 @@
+            )
+            return None
+
+        # Load the first matching dataset as a GeoDataFrame. Only shapefile
+        # formats are loadable; other formats would need a different reader.
+        try:
+            return client.load_shapefile(datasets[0], **kwargs)
+        except (OSError, ValueError) as exc:
+            raise BoundaryFetchError(
+                f"RDHProvider: failed to load {level} shapefile for state={state}, "
+                f"year={year}, format={fmt}"
+            ) from exc
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)